### PR TITLE
HDDS-5966. [HTTPFSGW] Update module doc, and place it in Ozone project docs

### PR DESCRIPTION
**Pull request by dombizita**
## What changes were proposed in this pull request?

Added proper documentation for the HttpFS gateway.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5966

## How was this patch tested?

Not really needed, CI on my fork: https://github.com/dombizita/ozone/actions/runs/4105032540

### CODE REVIEW

**Comment by umamaheswararao** [comment]
@DaveTeng0 @swamirishi

**Review by DaveTeng0 — APPROVED** [review]
looks good to me~

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+Go to the `compose/ozone` directory and start the server:
+
+```bash
+docker-compose up -d --scale datanode=3
+```
+
+You can now see the HttpFS gateway in docker with the name `ozone_httpfs`.
+HttpFS HTTP web-service API calls are HTTP REST calls that map to an Ozone file system operation. For example, using the `curl` Unix command.
+
+E.g. in the docker cluster you can execute commands these:
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:59]: I guess "like" is missing from this part: "commands like these"?

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+
+* `$ curl 'http://httpfs-host:14000/webhdfs/v1/user/foo/README.txt?op=OPEN&user.name=foo'` returns the contents of the `/user/foo/README.txt` key.
+
+
+## Supported operations
+
+These are the WebHDFS REST API operations that are supported/unsupported in Ozone.
+
+### File and Directory Operations
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:71]: For the following tables I would like to suggest to use the following terms:
- supported
- not implemented in Ozone
- not implemented in Ozone FileSystem API

For this to be meaningful, we should add the note about the internals I have suggested earlier.

**Review by fapifta — COMMENTED** [review]
Hi @dombizita thank you for working on the documentation piece for HttpFS, I have added some comments around wording. Probably I am not 100% right, it would be nice to get a native speaker to review as well though. :)

File: hadoop-hdds/docs/content/interface/HttpFS.md (hunk 2 above)
Comment (dombizita) [reply hadoop-hdds/docs/content/interface/HttpFS.md:71]: > we should add the note about the internals I have suggested earlier.
I'm not sure I got this part, what note do you think of?

**Comment by dombizita** [comment]
thanks for the review @fapifta, I addressed them, I had only one question regarding your last comment.

File: hadoop-hdds/docs/content/design/httpfs.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
+-->
+
+# Abstract
+
+With the HttpFS interface implementation, called Ozone HttpFS, we can provide a REST API - and with that, 
+easy integration for Ozone - for those tools that do not have an easy integration/migration to use the S3 
+interface or the FileSystem API.
```
Comment (smengcl) [inline hadoop-hdds/docs/content/design/httpfs.md:27]: CMIIW

```suggestion
Ozone HttpFS provides an HttpFS-compatible REST API interface to enable applications that are designed to use [HttpFS](https://hadoop.apache.org/docs/stable/hadoop-hdfs-httpfs/index.html) to interact and integrate with Ozone.
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+Ozone HttpFS can be used to integrate Ozone with other tools via REST API.
+
+## Introduction
+
+Ozone HttpFS is forked from the HDFS HttpFS endpoint implementation ([HDDS-5448](https://issues.apache.org/jira/browse/HDDS-5448)). It is added as a separate role to Ozone, like S3G.
```
Comment (smengcl) [inline hadoop-hdds/docs/content/interface/HttpFS.md:31]:
```suggestion
Ozone HttpFS is forked from the HDFS HttpFS endpoint implementation ([HDDS-5448](https://issues.apache.org/jira/browse/HDDS-5448)). It is added as a separate role to Ozone, similar to [S3 Gateway]({{< ref "design/s3gateway.md" >}}). 
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+Ozone HttpFS can be used to integrate Ozone with other tools via REST API.
+
+## Introduction
+
+Ozone HttpFS is forked from the HDFS HttpFS endpoint implementation ([HDDS-5448](https://issues.apache.org/jira/browse/HDDS-5448)). It is added as a separate role to Ozone, like S3G. 
+
+HttpFS is a service that provides a REST HTTP gateway supporting File System operations (read and write). It is interoperable with the **webhdfs** REST HTTP API.
+
+HttpFS can be used to access data on an Ozone cluster behind of a firewall (the HttpFS service acts as a gateway and is the only system that is allowed to cross the firewall into the cluster).
```
[the remark this answers is not in the document]
Comment (smengcl) [reply hadoop-hdds/docs/content/interface/HttpFS.md:35]: I do realize this is copied straight from https://hadoop.apache.org/docs/stable/hadoop-hdfs-httpfs/index.html but I can't help but reframed it a bit to my liking.

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+HttpFS can be used to access data in Ozone using HTTP utilities (such as curl and wget) and HTTP libraries Perl from other languages than Java.
+
+The **webhdfs** client FileSystem implementation can be used to access HttpFS using the Ozone filesystem command line tool (`ozone fs`) as well as from Java applications using the Hadoop FileSystem Java API.
+
+HttpFS has built-in security supporting Hadoop pseudo authentication and Kerberos SPNEGO and other pluggable authentication mechanisms. It also provides Hadoop proxy user support.
+
+
+## Getting started
+
+HttpFS service itself is a Jetty based web-application that uses the Hadoop FileSystem API to talk to the cluster, it is a separate service which provides access to Ozone via a REST API. It should be started additionally to the regular Ozone components.
```
Comment (smengcl) [inline hadoop-hdds/docs/content/interface/HttpFS.md:46]:
```suggestion
HttpFS service itself is a Jetty based web-application that uses the Hadoop FileSystem API to talk to the cluster, it is a separate service which provides access to Ozone via REST APIs. It should be started in addition to other regular Ozone components.
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+You can start a docker based cluster, including the HttpFS gateway from the release package.
```
Comment (smengcl) [inline hadoop-hdds/docs/content/interface/HttpFS.md:48]:
```suggestion
To try it out, you can start a Docker Compose cluster that has an HttpFS gateway.
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+Go to the `compose/ozone` directory and start the server:
```
Comment (smengcl) [inline hadoop-hdds/docs/content/interface/HttpFS.md:50]:
```suggestion
Extract the release tarball, go to the `compose/ozone` directory and start the cluster:
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+You can/should find now the HttpFS gateway in docker with the name `ozone_httpfs`.
+HttpFS HTTP web-service API calls are HTTP REST calls that map to an Ozone file system operation. For example, using the `curl` Unix command.
+
+E.g. in the docker cluster you can execute commands like these:
+
+* `curl -i -X PUT "http://httpfs:14000/webhdfs/v1/vol1?op=MKDIRS&user.name=hdfs"` creates a volume called `vol1`.
+
+
+* `$ curl 'http://httpfs-host:14000/webhdfs/v1/user/foo/README.txt?op=OPEN&user.name=foo'` returns the contents of the `/user/foo/README.txt` key.
```
Comment (smengcl) [inline hadoop-hdds/docs/content/interface/HttpFS.md:64]:
```suggestion
* `$ curl 'http://httpfs-host:14000/webhdfs/v1/user/foo/README.txt?op=OPEN&user.name=foo'` returns the content of the key `/user/foo/README.txt`.
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+
+
+## Supported operations
+
+These are the WebHDFS REST API operations that are supported/unsupported in Ozone.
```
Comment (smengcl) [inline hadoop-hdds/docs/content/interface/HttpFS.md:69]:
```suggestion
Here are the tables of WebHDFS REST APIs and their state of support in Ozone.
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+### File and Directory Operations
+
+Operation                       |      Support
+--------------------------------|---------------------
+Create and Write to a File      | supported
+Append to a File                | not implemented in Ozone FileSystem API
+Concat File(s)                  | not implemented in Ozone FileSystem API
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:77]:
```suggestion
Concat File(s)                  | not implemented in Ozone
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+Open and Read a File            | supported
+Make a Directory                | supported
+Create a Symbolic Link          | unsupported
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:80]:
```suggestion
Create a Symbolic Link          | not implemented in Ozone
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+
+### Other File System Operations
+
+Operation                             |      Support
+--------------------------------------|---------------------
+Get Content Summary of a Directory    | supported
+Get Quota Usage of a Directory        | supported
+Set Quota                             | not implemented in Ozone
+Set Quota By Storage Type             | not implemented in Ozone
+Get File Checksum                     | not implemented in Ozone
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:98]:
```suggestion
Get File Checksum                     | unsupported (to be fixed)
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+Get Home Directory                    | supported
+Get Trash Root                        | supported
+Set Permission                        | supported
+Set Owner                             | supported
+Set Replication Factor                | supported
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:103]:
```suggestion
Set Replication Factor                | not implemented in Ozone FileSystem API
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+--------------------------------------|---------------------
+Get Content Summary of a Directory    | supported
+Get Quota Usage of a Directory        | supported
+Set Quota                             | not implemented in Ozone
+Set Quota By Storage Type             | not implemented in Ozone
+Get File Checksum                     | not implemented in Ozone
+Get Home Directory                    | supported
+Get Trash Root                        | supported
+Set Permission                        | supported
+Set Owner                             | supported
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:102]:
```suggestion
Set Owner                             | not implemented in Ozone FileSystem API
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+Operation                             |      Support
+--------------------------------------|---------------------
+Get Content Summary of a Directory    | supported
+Get Quota Usage of a Directory        | supported
+Set Quota                             | not implemented in Ozone
+Set Quota By Storage Type             | not implemented in Ozone
+Get File Checksum                     | not implemented in Ozone
+Get Home Directory                    | supported
+Get Trash Root                        | supported
+Set Permission                        | supported
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:101]:
```suggestion
Set Permission                        | not implemented in Ozone FileSystem API
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+### Other File System Operations
+
+Operation                             |      Support
+--------------------------------------|---------------------
+Get Content Summary of a Directory    | supported
+Get Quota Usage of a Directory        | supported
+Set Quota                             | not implemented in Ozone
+Set Quota By Storage Type             | not implemented in Ozone
+Get File Checksum                     | not implemented in Ozone
+Get Home Directory                    | supported
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:99]:
```suggestion
Get Home Directory                    | unsupported (to be fixed)
```

File: hadoop-hdds/docs/content/interface/HttpFS.md [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+Get Trash Root                        | supported
+Set Permission                        | supported
+Set Owner                             | supported
+Set Replication Factor                | supported
+Set Access or Modification Time       | supported
```
Comment (fapifta) [inline hadoop-hdds/docs/content/interface/HttpFS.md:104]:
```suggestion
Set Access or Modification Time       | not implemented in Ozone FileSystem API
```

**Comment by fapifta** [comment]
Thank you @dombizita to address the comments, I have restarted the failed CI check, let's see, +1 to commit once it gets green ;)

**Pull request merged**